presto-main/src/main/java/com/facebook/presto/operator/OperatorContext.java

```diff
@@ -344,14 +344,14 @@ public void transferMemoryToTaskContext(long taskBytes)
         LocalMemoryContext taskLocalMemoryContext = driverContext.getPipelineContext().getTaskContext().localUserMemoryContext();
         try {
             operatorMemoryContext.localUserMemoryContext().transferOwnership(taskLocalMemoryContext);
-            taskLocalMemoryContext.addBytes(taskBytes - bytes);
+            taskLocalMemoryContext.setBytes(taskLocalMemoryContext.getBytes() + taskBytes - bytes);
```

Comment:
If you are doing `x.setBytes(x.getBytes() + y)`, there is really not much different than having `addBytes`.

In addition, this breaks thread safety.

Reading through your code, there are 3 situations where you used `getBytes`:

* tests
* peak memory tracking
* update(get() + ...)